WalletWasabi.Fluent/ViewModels/Wallets/Send/LabelSelectionViewModel.cs

```diff
@@ -167,10 +167,26 @@ private async Task<(bool, Pocket[] pockets)> TryGetBestKnownByRecipientPocketsWi
 			.Union(_hiddenIncludedPockets)
 			.ToArray();
 
-	public async Task ResetAsync(Pocket[] pockets)
+	public async Task ResetAsync(Pocket[] pockets, List<SmartCoin>? coinsToExclude = null)
 	{
 		_allPockets = pockets;
 
+		if (coinsToExclude is not null)
+		{
+			var pocketsWithoutExcludedCoins = new List<Pocket>();
+			var coinjoiningPockets = pockets.Where(pocket => pocket.Coins.Any(coinsToExclude.Contains)).ToArray();
+
+			foreach (var pocket in coinjoiningPockets)
+			{
+				pocketsWithoutExcludedCoins.Add(new Pocket((pocket.Labels, new CoinsView(pocket.Coins.Except(coinsToExclude)))));
+			}
+
+			if (await IsPocketEnoughAsync(pocketsWithoutExcludedCoins.ToArray()))
+			{
+				_allPockets = pocketsWithoutExcludedCoins.ToArray();
+			}
```

Comment:
The logic here seems to be wrong. 
- `coinjoiningPockets` has all the pocket that has at least 1 coin that is CJing.
- You create new pockets from those pockets but without the coins that are CJing.
- if they are enough, you override `_allPockets`.

So initially if you had a pocket that didn't have any coin CJing, you will lose it and that pocket won't be stored in `_allPockets`.

Reply:
Not tested, but this should do the work:
```c#
var pocketsWithoutExcludedCoins =
	pockets
		.Select(x => new Pocket((x.Labels, new CoinsView(x.Coins.Except(coinsToExclude)))))
		.ToArray();
```

So you could do:
```c#
var pocketsWithoutExcludedCoins =
	pockets
		.Select(x => new Pocket((x.Labels, new CoinsView(x.Coins.Except(coinsToExclude)))))
		.ToArray();

if (await IsPocketEnoughAsync(pocketsWithoutExcludedCoins))
{
	_allPockets = pocketsWithoutExcludedCoins;
}
```

Reply:
Oh shit, you are complete right. I will test it and push it.